File: backend/src/main/java/com/art/auction/model/Auction.java

package com.art.auction.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Document(collection = "auctions")
public class Auction {
    
    @Id
    private String id;
    
    private String name;
    
    private String description;
    
    private String sellerId;
    
    private String sellerName;
    
    private double minBid;
    
    private double currentBid;
    
    private String currentBidderId;
    
    private String currentBidderName;
    
    private String imageUrl;
    
    private List<Bid> bids = new ArrayList<>();
    
    private AuctionStatus status = AuctionStatus.ACTIVE;
    
    private LocalDateTime endTime;
    
    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
    
    // Enum for auction status
    public enum AuctionStatus {
        ACTIVE,
        ENDED,
        CANCELLED
    }
}

File: src/types/auction.ts

export interface Auction {
  id: string;
  name: string;
  description: string;
  seller_id: string;
  seller_name: string;
  min_bid: number;
  current_bid: number;
  current_bidder_id?: string;
  current_bidder_name?: string;
  image_url?: string;
  status: string; // Changed from specific union to string to match Supabase
  category?: string;
  end_time: string;
  created_at: string;
  updated_at: string;
}

export interface AuctionRequest {
  name: string;
  description: string;
  minBid: number;
  durationHours: number;
  category?: string;
}

export interface BidRequest {
  auctionId: string;
  amount: number;
}
